src/mil_common/electrical_protocol_cpp/include/electrical_protocol_cpp/packet.h

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+#include <string>
+#include <memory>
+#include <vector>
+#include <deque>
+#include <exception>
+
+#include <cppystruct.h>
+// #include <cppystruct/format.h>
+
+namespace electrical_protocol
+{
+    class SerialDevice;
+
+    class Packet
+    {
+        public:
+        friend class SerialDevice;
+        Packet() = delete;
+        ~Packet() noexcept
+        {
+
+        }
+        
+        Packet(uint8_t classId, uint8_t subClassId) noexcept :id_({classId, subClassId})
+        {
+        }
+
+        explicit Packet(const std::pair<uint8_t, uint8_t>& id) noexcept :id_(id) 
+        {
+        }
+
+        Packet(Packet&& packet) noexcept : id_(packet.id_)
+        {
+            data_ = std::move(packet.data_);
+        }
+
+        Packet(const Packet& packet) noexcept : id_(packet.id_)
+        {
+            data_ = packet.data_;
+        }
+
+        inline Packet& operator= (Packet&& packet)
+        {
+            if(id_ != packet.id_)
+                throw std::runtime_error("= between two packets with different id is impossiable");
+            
+            data_ = std::move(packet.data_);
+            return *this;
+        }
+
+        inline Packet& operator= (Packet& packet)
+        {
+            if(id_ != packet.id_)
+                throw std::runtime_error("= between two packets with different id is impossiable");
+            
+            data_ = packet.data_;
+            
+            return *this;
+        }
+
+        inline uint8_t& operator[] (size_t index)
+        {
+            if(index + HEADER_LEN + TRAILER_LEN >= data_.size())
+                throw std::out_of_range("No enough data");
+
+            return data_[index + HEADER_LEN];
+        }
+
+        inline const uint8_t& operator[] (size_t index) const
+        {
+            if(index + HEADER_LEN + TRAILER_LEN >= data_.size())
+                throw std::out_of_range("No enough data");
+
+            return data_[index + HEADER_LEN];
+        }
```

Comment:
I don't think we really need these functions. Since we hope to be using these packets at a pretty high-level, we probably won't need to access specific indices of the packed data. If we do, we can just call `pack()` and check a specific index of that.